libraries/aws_states_state_machines.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsStatesStateMachines < AwsResourceBase
+  name 'aws_states_state_machines'
+  desc 'Verifies settings for a collection of AWS StepFunction State Machines'
+  example "
+    describe aws_states_state_machines do
+      its('count') { should eq 3 }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:arns, field: :arn)
+             .register_column(:names, field: :name)
+             .register_column(:types, field: :type)
+             .register_column(:tags, field: :tags)
+             .register_column(:status, field: :status)
+             .register_column(:log_levels, field: :log_level)
+             .register_column(:log_execution_datas, field: :log_execution_data)
+             .register_column(:log_execution_datas, field: :log_execution_data)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def fetch_data
+    state_machine_rows = []
+    catch_aws_errors do
+      @api_response = @aws.states_client.list_state_machines
```

Comment:
[list_state_machines](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/States/Client.html#list_state_machines-instance_method) is a pageable response, so you will have to build the logic to read the paginated requests